### PR TITLE
PARQUET-204: add parquet-schema directory support

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

**Comment by tsdeng** [comment]
This is good. +1
Thanks @nevillelyh

File: parquet-hadoop/src/main/java/parquet/hadoop/util/HiddenFileFilter.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+
+public class HiddenFileFilter implements PathFilter {
+  @Override
+  public boolean accept(Path p) {
+    return !p.getName().startsWith("_") && !p.getName().startsWith(".");
+  }
+}
```
Comment (julienledem) [inline parquet-hadoop/src/main/java/parquet/hadoop/util/HiddenFileFilter.java]: As this is stateless there could be one static instance of it.
HIDDEN_FILE_FILTER = new HiddenFileFilter();

**Comment by julienledem** [comment]
Thanks for cleaning up all this duplicated code.
+1

File: parquet-hadoop/src/main/java/parquet/hadoop/util/HiddenFileFilter.java (hunk 1 above)
Comment (rdblue) [inline parquet-hadoop/src/main/java/parquet/hadoop/util/HiddenFileFilter.java]: Agree

**Comment by nevillelyh** [comment]
I changed it to a static INSTANCE member.

File: parquet-tools/src/main/java/parquet/tools/command/ShowSchemaCommand.java [hunk 2]
```diff
@@ -68,7 +71,21 @@ public void execute(CommandLine options) throws Exception {
...
+    if (fs.isDirectory(path)) {
+      FileStatus[] statuses = fs.listStatus(path, HiddenFileFilter.INSTANCE);
+      if (statuses.length == 0) {
+        throw new RuntimeException("Directory " + path.toString() + " is empty");
+      }
+      file = statuses[0].getPath();
+    } else {
+      file = path;
+    }
+    metaData = ParquetFileReader.readFooter(conf, file);
```
Comment (liancheng) [inline parquet-tools/src/main/java/parquet/tools/command/ShowSchemaCommand.java]: Should we take schema evolution into account here? I.e., show the merged schemas of all part-files.

Comment (rdblue) [inline parquet-tools/src/main/java/parquet/tools/command/ShowSchemaCommand.java]: I don't think so. There's no guarantee that there is a single schema for the data and merging all of the schemas into one would be misleading: a union strategy can produce a schema that can't be satisfied (as a column projection) by any of the files. I think it's best to return one or all of the unique schemas, but this is already going slightly beyond what Parquet itself should be doing as a file format. Parquet reads and writes files, while Hive or Kite manages the data as a collection.

**Comment by rdblue** [comment]
Thanks @nevillelyh!

**Pull request closed without merging**